./v test vlib/v/tests/generics_chans_select_test.v fails

### Describe the bug

This is on FreeBSD-14.0-p4. I did a fresh install of v with git clone & gmake. There is a problem wit tcc so I removed thirdparty/tcc. Running the above test fails. Every other test succeeds.

### Reproduction Steps

Do a fresh install on a freebsd-14.0 machine (patch number doesn't matter). Make sure `boehm-gc-threaded` pkg is installed. Then `gmake`, remove `thirdparty/gcc` and then run the above test.

### Expected Behavior

The test should pass.

### Current Behavior

See below

### Possible Solution

No idea.

### Additional Information/Context

```
./v test vlib/v/tests/generics_chans_select_test.v
---- Testing... -----------------------------------------------------------------
 FAIL   724.968 ms vlib/v/tests/generics_chans_select_test.v
==================
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/v_1001/tsession_262e37012000_705016034380/generics_chans_select_test.17295752555952635124.tmp.c:25386:6: error: incompatible pointer to integer conversion assigning to 'bool' (aka 'unsigned char') from 'anon_fn_main__foo__bool' (aka 'unsigned char (*)(struct main__Foo)') [-Wint-conversion]
                        b = ( (*(anon_fn_main__foo__bool*)_t5.data));
                          ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/v_1001/tsession_262e37012000_705016034380/generics_chans_select_test.17295752555952635124.tmp.c:25387:4: warning: expression result unused [-Wunused-value]
                        e;
                        ^
/tmp/v_1001/tsession_262e37012000_705016034380/generics_chans_select_test.17295752555952635124.tmp.c:25452:6: error: incompatible pointer to integer conversion assigning to 'bool' (aka 'unsigned char') from 'anon_fn_main__bar__bool' (aka 'unsigned char (*)(struct main__Bar)') [-Wint-conversion]
                        b = ( (*(anon_fn_main__bar__bool*)_t5.data));
                          ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/v_1001/tsession_262e37012000_705016034380/generics_chans_select_test.17295752555952635124.tmp.c:25453:4: warning: expression result unused [-Wunused-value]
                        e;
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

---------------------------------------------------------------------------------
Failed command 1:    '/home/bakul/v/v'   -o '/tmp/v_1001/tsession_262e37012000_705016034380/01HJS780B8S78RYTHJSNTT2VPZ/generics_chans_select_test' '/home/bakul/v/vlib/v/tests/generics_chans_select_test.v'
Summary for all V _test.v files: 1 failed, 1 total. Runtime: 730 ms, on 1 job.
```

### V version

V 0.4.3 680b0d4

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.3 2dce525.680b0d4
OS: freebsd, 14.0-RELEASE-p4, FreeBSD 14.0-RELEASE-p4 #47 releng/14.0-n265400-4edf3b80733e: Tue Dec 19 15:35:46 PST 2023     xxx:/usr/obj/home/FreeBSD/releng-14/amd64.amd64/sys/GENERIC
Processor: 1 cpus, 64bit, little endian

getwd: /home/bakul/v
vexe: /home/bakul/v/v
vexe mtime: 2023-12-28 22:02:16

vroot: OK, value: /home/bakul/v
VMODULES: OK, value: /home/bakul/.vmodules
VTMP: OK, value: /tmp/v_1001

Git version: git version 2.42.0
Git vroot status: 680b0d46
.git/config present: true

CC version: FreeBSD clang version 16.0.6 (https://github.com/llvm/llvm-project.git llvmorg-16.0.6-0-g7cbf1a259152)
thirdparty/tcc: N/A
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/assign.v
+++ b/vlib/v/checker/assign.v
@@ -735,8 +735,7 @@ or use an explicit `unsafe{ a[..] }`, if you do not want a copy of the slice.',
 				}
 			}
 		}
-		if !is_blank_ident && right_sym.kind != .placeholder && left_sym.kind != .interface_
-			&& !right_type.has_flag(.generic) && !left_type.has_flag(.generic) {
+		if !is_blank_ident && right_sym.kind != .placeholder && left_sym.kind != .interface_ {
 			// Dual sides check (compatibility check)
 			c.check_expected(right_type_unwrapped, left_type_unwrapped) or {
 				// allow literal values to auto deref var (e.g.`for mut v in values { v = 1.0 }`)

--- a/vlib/v/checker/tests/assign_type_mismatch_with_generics_err.out
+++ b/vlib/v/checker/tests/assign_type_mismatch_with_generics_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/assign_type_mismatch_with_generics_err.vv:13:9: error: cannot assign to `b`: expected `bool`, not `fn (Bar) bool`
+   11 |     mut b := false
+   12 |     if f.f != none {
+   13 |         b = f.f or { panic(err) }
+      |               ^
+   14 |     } else {
+   15 |         b = true

--- a/vlib/v/checker/tests/assign_type_mismatch_with_generics_err.vv
+++ b/vlib/v/checker/tests/assign_type_mismatch_with_generics_err.vv
@@ -1,0 +1,28 @@
+type Fn[T] = fn (arg T) bool
+
+struct Foo[T] {
+	f ?Fn[T]
+}
+
+@[params]
+struct Bar {}
+
+fn (mut f Foo[T]) method(arg T) {
+	mut b := false
+	if f.f != none {
+		b = f.f or { panic(err) }
+	} else {
+		b = true
+	}
+	if b {
+	}
+}
+
+fn main() {
+	mut foo := Foo[Bar]{
+		f: fn (arg Bar) bool {
+			return true
+		}
+	}
+	foo.method(Bar{})
+}

--- a/vlib/v/tests/generics_chans_select_test.v
+++ b/vlib/v/tests/generics_chans_select_test.v
@@ -35,8 +35,8 @@ pub fn (mut ec EventController[T]) emit(e T, options EmitOptions) {
 	for i, w in ec.wait_fors {
 		mut b := false
 		if w.check != none {
-			b = (w.check or { panic(err) })
-			e
+			func := w.check or { panic(err) }
+			b = func(e)
 		} else {
 			b = true
 		}

